### PR TITLE
fix: set postgres timeout args via query

### PR DESCRIPTION
**Pull request by jhlwscom**
## Summary & Motivation

AWS RDS Proxy does not support cli options, so setting any mandatory connection options via statement improves compatibility.
Any custom options are still passed through as prior

## How I tested changes

- dagster_postgres pytests
- `dagster dev` and connect via rds proxy and directly, storage loads as expected

Resolves: #11762

### CODE REVIEW

**Comment by danielgafni** [comment]
Would be very happy if this gets merged! I'm experiencing the same issue. @garethbrickman have you guys noticed this PR?

**Comment by garethbrickman** [comment]
@danielgafni Got this one the agenda to review this Wednesday, thanks for the reminder! 👍

**Comment by telderfts** [comment]
@garethbrickman Have you had a chance to review this? Quite a few people would like to use RDS proxies for connection pooling, etc and this is blocking that.

**Comment by jhlwscom** [comment]
This iteration passes `python -m pytest python_modules/libraries/dagster-postgres/dagster_postgres_tests`, time to see what buildkite says.

**Comment by gibsondan** [comment]
@jhlwscom seeing some test failures here in the dagster-postgres package that look potentially relevant: "TypeError: DefaultEngineStrategy.create() missing 1 required positional argument: 'name_or_url'"

File: python_modules/libraries/dagster-postgres/dagster_postgres/event_log/event_log.py [hunk 1]
```diff
@@ -112,18 +113,20 @@ def _init_db(self) -> None:
...
+            "pool_recycle": pool_recycle,
+        }
         existing_options = self._engine.url.query.get("options")
-        timeout_option = pg_statement_timeout(statement_timeout)
         if existing_options:
-            options = f"{timeout_option} {existing_options}"
-        else:
-            options = timeout_option
-        self._engine = create_engine(
-            self.postgres_url,
```
Comment (gibsondan) [inline python_modules/libraries/dagster-postgres/dagster_postgres/event_log/event_log.py:122]: it seems like at least on sqlalchemy 1, this needs to be a positional argument (or a 'name_or_url' kwarg)

**Review by gibsondan — CHANGES_REQUESTED** [review]
That should hopefully be a quick change to make the argument positional instead of a kwarg, after which this looks good to me. Be sure to run 'make ruff' and 'make pyright' when you resubmit. Apologies for the delay on getting this looked at.

**Comment by jhlwscom** [comment]
Sorry for the delay on that tweak, was camping.

**Review by gibsondan — APPROVED** [review]
thanks!

**Pull request merged**